ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -340,24 +344,111 @@ azure_object_size(
     return 0;
 }
 
+// File open for the Azure storage source.
 static int
 azure_file_open(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name,
   WT_FS_OPEN_FILE_TYPE file_type, uint32_t flags, WT_FILE_HANDLE **file_handlep)
 {
-    WT_UNUSED(file_system);
+    azure_file_system *azure_fs = reinterpret_cast<azure_file_system *>(file_system);
+
+    // Azure only supports opening the file in read only mode.
+    if ((flags & WT_FS_OPEN_READONLY) == 0 || (flags & WT_FS_OPEN_CREATE) != 0) {
+        std::cerr << "azure_file_open: read-only access required." << std::endl;
+        return EINVAL;
+    }
+
+    // Only data files should be opened; although this constraint can be relaxed in future.
+    if (file_type != WT_FS_OPEN_FILE_TYPE_DATA && file_type != WT_FS_OPEN_FILE_TYPE_REGULAR) {
+        std::cerr << "azure_file_open: only data file and regular types supported." << std::endl;
+        return EINVAL;
+    }
+
+    // Check if object exists.
+    bool exists;
+    int ret;
+    if ((ret = azure_fs->azure_conn->object_exists(std::string(name), exists)) != 0) {
```

Comment:
You would also need to check `!exists` here because if the file does not exist the function will still return 0

Reply:
yep I check if its false under this statement